packages/from-to-location-picker/src/index.js

```diff
@@ -1,59 +1,100 @@
-import React from "react";
+import { locationType } from "@opentripplanner/core-utils/lib/types";
+import LocationIcon from "@opentripplanner/location-icon";
 import PropTypes from "prop-types";
+import React, { Component } from "react";
 
-import LocationIcon from "@opentripplanner/location-icon";
 import { LocationPickerSpan, Button, FromToPickerSpan } from "./styled";
 
 const iconSize = "0.9em";
 
-const FromToLocationPicker = ({
-  fromText,
-  toText,
-  showIcons,
-  onFromClick,
-  onToClick
-}) => {
-  return (
-    <FromToPickerSpan>
-      <LocationPickerSpan>
-        {showIcons && <LocationIcon type="from" size={iconSize} />}
-        <Button onClick={onFromClick}>{fromText}</Button>
-      </LocationPickerSpan>
-      <LocationPickerSpan>
-        {showIcons && <LocationIcon type="to" size={iconSize} />}
-        <Button onClick={onToClick}>{toText}</Button>
-      </LocationPickerSpan>
-    </FromToPickerSpan>
-  );
-};
+class FromToLocationPicker extends Component {
+  onFromClick = () => {
+    const { location, onFromClick, setLocation } = this.props;
+    if (onFromClick) {
+      onFromClick();
+      return;
+    }
+    setLocation({
```

Comment:
Does the code that handles setLocation split into two different paths for 'from' and 'to'?

Reply:
This is somewhat of an artifact of extracting stuff from otp-react-redux. Here is the relevant action: https://github.com/opentripplanner/otp-react-redux/blob/906d9226553cbbdd83cf455986f199c5a6964f72/lib/actions/map.js#L38-L63